src/gleam/float.gleam

```diff
@@ -320,3 +335,50 @@ fn do_product(numbers: List(Float), initial: Float) -> Float {
     [x, ..rest] -> do_product(rest, x *. initial)
   }
 }
+
+/// Returns a uniform random number
+/// Thus where 0.0 =< value < 1.0
+///
+pub fn random_uniform() -> Float {
+  do_random_uniform()
+}
+
+if erlang {
+  /// Returns a random float uniformly distributed in the value range
+  /// 0.0 =< X < 1.0 and updates the state in the process dictionary.
+  /// See: <https://www.erlang.org/doc/man/rand.html#uniform-0>
+  ///
+  external fn do_random_uniform() -> Float =
+    "rand" "uniform"
+}
+
+if javascript {
+  external fn do_random_uniform() -> Float =
+    "../gleam_stdlib.mjs" "random_uniform"
+}
+
+/// Returns 0.0 if boundary_a and boundary_b are equal
+/// Based on:
+/// ```javascript
+/// return Math.random() * (max - min) + min; // The minimum is inclusive and the maximum is exclusive
+/// ```
+/// See: <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random#getting_a_random_number_between_two_values>
+///
+pub fn random_between(boundary_a: Float, boundary_b: Float) -> Float {
```

Comment:
Can we have an API like the one found in float with a `min` and `max` parameter? I think that is a bit clearer.

I think `random` would be a nice name for this function, and we can remove the `random_uniform` as this function can be used for that behaviour too with `random(0.0, 1.0)`

Reply:
I had it at `min` and `max` initially but felt that that was a lie, it is just two boundaries on a scale. If we use `min` and `max` we should IMHO use a Result type and validate that `min` < `max` and disallow `min == max`.

Reply:
I have unified the API to be both like `float.random()` where it does not matter if you put min or max in reverse order.